sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2616,4 +2617,66 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
     val e = intercept[AnalysisException](sql("SELECT nvl(1, 2, 3)"))
     assert(e.message.contains("Invalid number of arguments"))
   }
+
+  test("TRIM function-BOTH") {
+    val ae1 = intercept[ParseException]{
+      sql("select lower(BOTH 'S' FROM 'SS abc S')").head
```

Comment:
ltrim/rtrim can't using these options, right? We can use them as negative test for function names, and use trim for negative tests for option names.